ocaml/xapi-idl/storage/storage_interface.ml

```diff
@@ -354,6 +364,7 @@ module Errors = struct
     | Cancelled of string
     | Redirect of string option
     | Sr_attached of string
+    | Sr_unhealthy of sr_health
```

Comment:
I don't like that we can have `Sr_unhealthy(Healthy)` - what is that supposed to mean? I think the constructor should be `Sr_health` (it's too late to use `SR_health` because we have introduced the `Sr_` prefix already.

Reply:
Just to clarify, we can't have `Sr_unhealthy Healthy` as that will never (at least for now) be generated, although I agree with you that this is not enforced from the type system point of view

Reply:
As this is an error type, I think `Sr_unhealthy` works as a constructor, with the parameter to provide further detail.

Reply:
I understand how this came into existence; as long as `Healthy` can't (or should) not show up as a detail this is clear enough.